lib/db_utils.go

```diff
@@ -1643,6 +1731,546 @@ func DBGetAllUserGroupEntries(handle *badger.DB, ownerPublicKey []byte) ([]*Mess
+				accessGroupOwnerPublicKeyFromKey, accessGroupOwnerPublicKey)
+		}
+
+		accessGroupId := NewAccessGroupId(&accessGroupOwnerPublicKey, accessGroupKeyName.ToBytes())
+		accessGroupIds = append(accessGroupIds, accessGroupId)
+	}
+	return accessGroupIds, nil
+}
+
+func DBPutAccessGroupEntry(db *badger.DB, snap *Snapshot, blockHeight uint64, accessGroupEntry *AccessGroupEntry) error {
+	var err error
+	err = db.Update(func(txn *badger.Txn) error {
+		return DBPutAccessGroupEntryWithTxn(txn, snap, blockHeight, accessGroupEntry)
+	})
+	if err != nil {
+		return errors.Wrapf(err, "DBPutAccessGroupEntry: Problem putting access group entry")
+	}
+	return nil
+}
+
+func DBPutAccessGroupEntryWithTxn(txn *badger.Txn, snap *Snapshot, blockHeight uint64, accessGroupEntry *AccessGroupEntry) error {
+	prefix := _dbKeyForAccessGroupEntry(*accessGroupEntry.AccessGroupOwnerPublicKey, *accessGroupEntry.AccessGroupKeyName)
+	if err := DBSetWithTxn(txn, snap, prefix, EncodeToBytes(blockHeight, accessGroupEntry)); err != nil {
+		return errors.Wrapf(err, "DBPutAccessGroupEntryWithTxn: Problem putting access group entry")
+	}
+
+	return nil
+}
+
+func DBDeleteAccessGroupEntry(db *badger.DB, snap *Snapshot, accessGroupOwnerPublicKey PublicKey,
+	accessGroupKeyName GroupKeyName) error {
+	var err error
+	err = db.Update(func(txn *badger.Txn) error {
+		return DBDeleteAccessGroupEntryWithTxn(txn, snap, accessGroupOwnerPublicKey, accessGroupKeyName)
+	})
+	if err != nil {
+		return errors.Wrapf(err, "DBDeleteAccessGroupEntry: Problem deleting access group entry")
+	}
+	return nil
+}
+
+func DBDeleteAccessGroupEntryWithTxn(txn *badger.Txn, snap *Snapshot, accessGroupOwnerPublicKey PublicKey,
+	accessGroupKeyName GroupKeyName) error {
+
+	prefix := _dbKeyForAccessGroupEntry(accessGroupOwnerPublicKey, accessGroupKeyName)
+
+	if err := DBDeleteWithTxn(txn, snap, prefix); err != nil {
+		return errors.Wrapf(err, "DBDeleteAccessGroupEntryWithTxn: Problem deleting access group entry")
+	}
+
+	return nil
+}
+
+// -------------------------------------------------------------------------------------
+// Access group member entry db functionality
+// PrefixAccessGroupMembershipIndex
+// <prefix, AccessGroupMemberPublicKey, AccessGroupOwnerPublicKey, AccessGroupKeyName> -> <AccessGroupMemberEntry>
+// -------------------------------------------------------------------------------------
+
+func _dbKeyForAccessGroupMemberEntry(
+	accessGroupMemberPublicKey PublicKey, accessGroupOwnerPublicKey PublicKey, accessGroupKeyName GroupKeyName) []byte {
+
+	prefixCopy := append([]byte{}, Prefixes.PrefixAccessGroupMembershipIndex...)
+	prefixCopy = append(prefixCopy, accessGroupMemberPublicKey.ToBytes()...)
+	prefixCopy = append(prefixCopy, accessGroupOwnerPublicKey.ToBytes()...)
+	prefixCopy = append(prefixCopy, accessGroupKeyName.ToBytes()...)
+	return prefixCopy
+}
+
+func _dbSeekPrefixForAccessGroupMemberEntry(accessGroupMemberPublicKey PublicKey) []byte {
```

Comment:
might we want a seek function for accessGroupMember + groupOwner? "what are all my group chat with User X" use case. not critical now, but food for thought.